PhotoBooth/Camera Page/CameraViewController.swift

```diff
@@ -172,9 +179,51 @@ class CameraViewController: UIViewController {
             }
         }
     }
-    
+
+    @objc func flashMode() {
+        switch viewModel.flashType {
+        case .on:
+            UIView.animate(withDuration: 0.1,
+                           animations: {
+                            self.flashButton.transform = CGAffineTransform(scaleX: 0.6, y: 0.6)
+                            self.viewModel.toggleFlash()
+            },
+                           completion: { _ in
+                            UIView.animate(withDuration: 0.1) {
+                                print(self.viewModel.captureSession.getFlashMode().rawValue)
+                                self.flashButton.setImage(UIImage(named: "flash_off"), for: .normal)
+                                self.flashButton.transform = CGAffineTransform.identity
+                            }
+            })
+        case .off:
+            UIView.animate(withDuration: 0.1,
+                           animations: {
+                            self.flashButton.transform = CGAffineTransform(scaleX: 0.6, y: 0.6)
+                            self.viewModel.toggleFlash()
+            },
+                           completion: { _ in
+                            UIView.animate(withDuration: 0.1) {
+                                print(self.viewModel.captureSession.getFlashMode().rawValue)
+                                self.flashButton.setImage(UIImage(named: "flash_on"), for: .normal)
```

Comment:
these two animation blocks are the same. the only difference is the image. Can you use the the switch statement to choose an image and then do the animation in one go?

```swift
var flashImage: UIImage?
switch {
  set flashImage
}

UIView.animate(.... {
 completion: {

       self.flashButton.setImage(flashImage)
       ....
   }
})
```

Reply:
done